Infragraph LA docs update [IPE-1505] (#2710)
hashicorp/web-unified-docs: content/hcp-docs/content/docs/infragraph/connections/azure.mdx

---
page_title: Connect Infragraph to Azure
description: |-
  The Azure connection connects your Microsoft Azure data to the resource graph. Learn how to set up the connection, sync data, and view Azure data in Infragraph.
---

# Connect Infragraph to Azure

The Azure connection connects your Microsoft Azure data to the resource graph.

@include 'beta-feature.mdx'

## Set up the connection

Complete the following steps to establish a new Azure connection:

1. From the Resource graph **Overview** at the top right, click **Create Connection**.
1. Select **Microsoft Azure** from the list of data sources.
1. Click **Select Microsoft Azure**.

> 1. Under **Connection target**, enter your **Azure Tenant ID**. This identifies the Microsoft Entra tenant that Infragraph connects to. To find your Tenant ID in the Azure portal, navigate to **Microsoft Entra ID** and copy the **Tenant ID** field on the **Overview** page.

Reviewer: ```suggestion
1. Under **Connection target**, enter your **Azure Tenant ID**. This identifies the Microsoft Entra tenant that Infragraph connects to. To find your Tenant ID, navigate to **Microsoft Entra ID** in the Azure portal and copy the **Tenant ID** field on the **Overview** page.
```

1. Click **View Terraform module in Registry** to review the Azure App Registration, Federated Identity Credential, and Role Assignment requirements.
1. Copy the generated module block and add it to your HCP Terraform or Terraform Enterprise linked VCS repository.
1. Configure your Azure credentials. Refer to the [Azure Terraform provider authentication](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs#authenticating-to-azure) documentation for more information.
1. Run a Terraform apply operation to create the required resources in your Azure environment.
1. Copy the `application_id` output value and paste it into the **Application (client) ID** field in the connection setup page.

> 1. We recommend you subscribe to the module GitHub repository to be notified of changes to ensure your connections continue to work.

Reviewer: Same comment as in `aws.mdx` for this recommendation

1. Click the **Provide connection details** button.
1. On the **Provide connection details** page, choose one of the following **Sync scopes**:
    - **Management Group**: Infragraph syncs all accounts in the organization. This is best for organizations using Azure Management Groups. Enter the **Management Group ID**. Infragraph syncs all subscriptions under this management group. This option requires `Management Group Reader` and `Reader` roles at the management group scope.

>     - **Specific Subscriptions**: Infragraph only syncs data from the specified accounts. This is best for granular control. Enter the **Subscription IDs** for Infragraph to sync. You can enter IDs manually separated by commas or newlines, or click **Upload CSV** to import up to 100 accounts in bulk.

Reviewer: ```suggestion
    - **Specific Subscriptions**: Infragraph only syncs data from the specified accounts. This is best for granular control. Enter the **Subscription IDs** for Infragraph to sync. You can manually enter IDs separated by commas or new lines or click **Upload CSV** to import up to 100 accounts in bulk.
```

1. Click the **Create connection** button.

Infragraph only requires read-only permissions. The Terraform module assigns the Azure Reader role at the appropriate scope level. Some resource types, such as Key Vault secrets, may require additional reader roles, such as Key Vault Reader, to discover metadata. Refer to [Troubleshooting](#troubleshooting) for more information.

## Sync Azure data

@include 'infragraph/sync-data.mdx'

## View Azure data

To find Azure data connected to your resource graph, complete any of the following actions:

- Go to the **Inventory** page. Click an individual resource type to view it in the graph explorer.

> - Query resources with any of the existing queries that include Azure resources.

Reviewer: ```suggestion
- Query resources with a saved query that includes Azure resources.
```

- Build the query from scratch using Azure resources in the query editor.

> ## Troubleshooting

Reviewer: This whole section is about troubleshooting KV discovery issues.  To avoid having a general "troubleshooting" section, you could make this "Troubleshooting Key Vault secret metadata discovery issues".  Or find a different word than "troubleshooting"

If the Infragraph connector finds Azure Key Vaults using the legacy **Access Policies** model, the **Sync status** shows as **Partially successful**. This is because Access Policies are configured per vault and do not honor RBAC role assignments of the Infragraph connector.

We recommend either of the following options to resolve this issue.

#### Migrate to Azure RBAC

Complete the following steps to migrate your Key Vault to use Azure RBAC instead of Access Policies:

1. In the Azure Portal, navigate to your Key Vault.
1. Go to **Access configuration**.
1. Change the permission model to **Azure role-based access control**.
1. Click **Save**.

Refer to the [Azure Key Vault RBAC migration guide](https://learn.microsoft.com/en-us/azure/key-vault/general/rbac-migration) for detailed instructions.

#### Add Access Policy for each vault

If you cannot migrate to RBAC, manually add an Access Policy granting **List** permission under **Secret Permissions** for the connector service principal on each Key Vault that uses Access Policies.

You can manually configure this using the Azure Portal or automate it using the following Azure CLI command or Terraform configuration.

> You only need to grant the `List` permission under Secret Permissions. The Infragraph connector does not require permissions to read secret values.

Reviewer: Worth adding "Follow the principle of least privilege by _NOT_ granting permissions to read secret values to the Infragraph connector"?

Or similar.

Just something so nobody over-permissions?
